validate/data.go

```diff
@@ -24,36 +33,59 @@ func EntryReader(r dsio.EntryReader) ([]jsonschema.ValError, error) {
 		return nil, fmt.Errorf("error allocating data buffer: %s", err.Error())
 	}
 
+	validated := false
+
 	err = dsio.EachEntry(r, func(i int, ent dsio.Entry, err error) error {
 		if err != nil {
 			return fmt.Errorf("error reading row %d: %s", i, err.Error())
 		}
+
 		err = buf.WriteEntry(ent)
 		if err != nil {
 			return fmt.Errorf("error writing row %d: %s", i, err.Error())
 		}
+		validated = false
+
+		if (i+1)%batchSize == 0 {
+			validated = true
+			if e := buf.Close(); e != nil {
```

Comment:
The behavior here, repeated after the end of the loop, could be moved into a local function var, called `flushBatch` or something, to avoid duplication and aid in clarity.